arch/posix/CMakeLists.txt

```diff
@@ -1,12 +1,23 @@
 # SPDX-License-Identifier: Apache-2.0
 
-if (CONFIG_64BIT)
-	# some gcc versions fail to build without -fPIC
-	zephyr_compile_options(-m64 -fPIC)
-	zephyr_link_libraries(-m64)
-else ()
-	zephyr_compile_options(-m32)
-	zephyr_link_libraries(-m32)
+# @Intent: Detect native posix architecture (see 'man 1 arch')
+execute_process(COMMAND arch OUTPUT_VARIABLE BUILD_ARCH OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+# @Intent: Set any *necessary* arch-specific compile / linker options so native_posix Just Works (TM)
+if (${BUILD_ARCH} STREQUAL "aarch64")
+    # The -fPIC option is required for case 1, but does not hurt case 2
+    # case 1: (64-bit OS, 64-bit userspace)
+    # case 2: (64-bit OS, 32-bit userspace)
+    zephyr_compile_options(-fPIC)
```

Comment:
I cannot test it, but I really doubt this will work for native_poxis(_32) as is. So I'd add a if !CONFIG_64BIT => error

Reply:
which is probably why he safeguarded the compile option with `if (CONFIG_64BIT)`

Reply:
This comment was related to a completely different content of this PR around this line, and is not applicable anymore as the PR is now.

Reply:
got it.